Geometry/EcalCommonData/plugins/DDEcalEndcapAlgo.cc

```diff
@@ -1,23 +1,205 @@
-
 //////////////////////////////////////////////////////////////////////////////
 // File: DDEcalEndcapAlgo.cc
 // Description: Geometry factory class for Ecal Barrel
 ///////////////////////////////////////////////////////////////////////////////
 
-#include <cmath>
-#include <algorithm>
-
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/Math/interface/GeantUnits.h"
 #include "DetectorDescription/Core/interface/DDLogicalPart.h"
 #include "DetectorDescription/Core/interface/DDSolid.h"
 #include "DetectorDescription/Core/interface/DDCurrentNamespace.h"
-#include "Geometry/EcalCommonData/interface/DDEcalEndcapAlgo.h"
-#include "CLHEP/Units/GlobalSystemOfUnits.h"
-
-#include <CLHEP/Geometry/Transform3D.h>
+#include "DetectorDescription/Core/interface/DDTypes.h"
+#include "DetectorDescription/Core/interface/DDName.h"
+#include "DetectorDescription/Core/interface/DDAlgorithm.h"
+#include "DetectorDescription/Core/interface/DDMaterial.h"
+#include "DetectorDescription/Core/interface/DDSplit.h"
+#include "DetectorDescription/Core/interface/DDTransform.h"
+#include "Geometry/CaloGeometry/interface/EcalTrapezoidParameters.h"
+#include "CLHEP/Geometry/Transform3D.h"
 
 // Header files for endcap supercrystal geometry
-#include "Geometry/EcalCommonData/interface/DDEcalEndcapTrap.h"
+#include "DDEcalEndcapTrap.h"
+
+#include <cmath>
+#include <algorithm>
+#include <map>
+#include <string>
+#include <vector>
+
+using namespace geant_units::operators;
+
+class DDEcalEndcapAlgo : public DDAlgorithm {
+public:
+  typedef EcalTrapezoidParameters Trap;
+  typedef HepGeom::Point3D<double> Pt3D;
+  typedef HepGeom::Transform3D Tf3D;
+  typedef HepGeom::ReflectZ3D RfZ3D;
+  typedef HepGeom::Translate3D Tl3D;
+  typedef HepGeom::Rotate3D Ro3D;
+  typedef HepGeom::RotateZ3D RoZ3D;
+  typedef HepGeom::RotateY3D RoY3D;
+  typedef HepGeom::RotateX3D RoX3D;
+
+  typedef CLHEP::Hep3Vector Vec3;
+  typedef CLHEP::HepRotation Rota;
```

Comment:
Can you change these CLHEP utilities to ROOT?

Reply:
I think it would be  good argument to involve ECal DPG to modernise their geometry code.